By: Eric Sandall
For: Source Mage GNU/Linux
Description: A quick fix to get gnupg-2.0.9 compiling with gcc 4.3.0 by obfuscating
the reference to a bit-field.
--
diff -Naur gnupg-2.0.9.orig/keyserver/gpgkeys_curl.c gnupg-2.0.9/keyserver/gpgkeys_curl.c
--- gnupg-2.0.9.orig/keyserver/gpgkeys_curl.c	2008-04-13 09:52:35.000000000 +0000
+++ gnupg-2.0.9/keyserver/gpgkeys_curl.c	2008-04-13 09:52:43.000000000 +0000
@@ -300,7 +300,14 @@
       curl_easy_setopt(curl,CURLOPT_VERBOSE,1);
     }
 
-  curl_easy_setopt(curl,CURLOPT_SSL_VERIFYPEER,opt->flags.check_cert);
+  /*
+   * GCC 4.3.0 fix for bit-fields
+   */
+  unsigned int check_cert = 0;
+  check_cert = opt->flags.check_cert;
+  curl_easy_setopt(curl,CURLOPT_SSL_VERIFYPEER,check_cert);
+  /* End GCC 4.3.0 fix */
+
   curl_easy_setopt(curl,CURLOPT_CAINFO,opt->ca_cert_file);
 
   if(proxy)
